pkg/credentials/utils.go

```diff
@@ -0,0 +1,32 @@
+package credentials
+
+import (
+	"encoding/base64"
+	"strings"
+)
+
+func ToBase64Url(buffer []byte) string {
+	encoded := base64.RawURLEncoding.EncodeToString(buffer)
+	encoded = strings.ReplaceAll(encoded, "=", "")
+	encoded = strings.ReplaceAll(encoded, "+", "-")
+	encoded = strings.ReplaceAll(encoded, "/", "_")
+
+	return encoded
+}
+
+func FromBase64Url(encoded string) ([]byte, error) {
+	padLength := 4 - (len(encoded) % 4)
+	if padLength < 4 {
+		encoded += strings.Repeat("=", padLength)
+	}
+
+	encoded = strings.ReplaceAll(encoded, "-", "+")
+	encoded = strings.ReplaceAll(encoded, "_", "/")
+
+	data, err := base64.StdEncoding.DecodeString(encoded)
```

Comment:
same here, you should be able to just use `RawURLEncoding.DecodeString` I think.